vision/segmentation/plane_segmentation.go

```diff
@@ -106,6 +161,12 @@ func SegmentPlane(ctx context.Context, cloud pc.PointCloud, nIterations int, thr
 		equations = append(equations, currentEquation)
 	}
 
+	return findBestEq(ctx, &cloud, nIterations, &equations, &pts, threshold)
+}
+
+func findBestEq(ctx context.Context, cloud *pc.PointCloud, nIterations int, equations *[][4]float64,
+	pts *[]r3.Vector, threshold float64,
```

Comment:
I don't think you need pointers to slices `*[][4]float64` and `*[]r3.Vector`. something like `[]r3.Vector` is already a pointer to an array -- so you don't have to worry about copying the whole slice of r3.Vector over. Just use `[][4]float64` and `[]r3.Vector`

Reply:
Same with `*pc.PointCloud`, PointCloud is already an interface, which means it is already a pointer. doing `*pc.PointCloud` is a pointer to a pointer.

Reply:
Got it, didn't know that applied for interfaces actually!